lib/datasets.go

```diff
@@ -198,11 +198,20 @@ func (r *DatasetRequests) Select(p *SelectParams, res *[]byte) (err error) {
 
 	encode := map[string]interface{}{}
 	for _, ref := range p.Refs {
-		if err = repo.CanonicalizeDatasetRef(r.repo.Repo, &ref); err != nil {
+		err = repo.CanonicalizeDatasetRef(r.repo.Repo, &ref)
+		if err != nil && err != repo.ErrNotFound {
 			log.Debug(err.Error())
 			return err
 		}
-
+		if err == repo.ErrNotFound {
```

Comment:
we've had an offline conversation about having a better / more explicit error response here, so I'm ok with `repo.ErrNotFound` for now. Let's get that one cleaned up ASAP.